src/lib.rs

```diff
@@ -279,6 +282,12 @@ impl UnixStream {
         self.io.get_ref().peer_addr()
     }
 
+    /// Returns effective credentials of the process which called `connect` or `socketpair`.
+    #[cfg(any(target_os = "linux", target_os = "macos"))]
+    pub fn peer_cred(&self) -> io::Result<UCred> {
```

Comment:
If we don't have a cfg here then we're claiming it's available on all unices, both in the code and in the documentation. At that point we need an assertion as such, which is to attempt to compile the code on all platforms.